### DISCUSSION TITLE
Ability to have loops within yaml and json to support collections

### DISCUSSION THREAD

**Discussion by kristofre** [Dynatrace/dynatrace-configuration-as-code#222] [Archive] [answered]
(I believe this might have been asked for before but couldn't find it anywhere)

Sometimes, you will have a case where you want to have a variable amount of items defined in the configuration.

For example, when defining a synthetic test, you can have multiple locations assigned to it:

test.json:
```json
"locations": [
      "{{ .locationone }}",
      "{{ .locationtwo }}"
    ],
```

test.yaml:
```
test-one:
- locationone: "LOCATIONONE-ID"
- locationtwo: "LOCATIONTWO-ID"
```

At the moment, you can define a variable for each location individually, but you would need a template for each test type per number of locations.

Therefor, it would be great if we could support collections, not unlike how helm supports it:

test.yaml:
```
test-one:
- locations:
    - "LOCATIONONE-ID"
    - "LOCATIONTWO-ID"
```

test.json:
```
"locations": [
     {{ locations .Values }}
         "{{ .value }}"
     {{ end }}
    ],
```

**Comment by kristofre** [comment]
btw, a current workaround is to supply the json directly in the variable:

test.json
``` json
"locations": [ {{ .locations }} ],
```

test.yaml
``` yaml
test-one:
- locations: "\"LOCATIONONE-ID\", \"LOCATIONTWO-ID\""
```

However that is not a clean solution and would get quite complex when the collection is also a multi attribute item.

**Reply by tatomir146** [reply]
This looks like it could be related to #165 as then we would anyways have to change yaml structure. Json would then to be modified using range function e.g.

```
{{range .Locations}}
     {{.Name}}
{{end}}
```

**Comment by tatomir146** [comment, marked as answer]
With monaco 2.0 this will change. More complex data structure will be possible.
